db/attachment_test.go

```diff
@@ -1359,3 +1360,131 @@ func TestMigrateBodyAttachmentsMergeConflicting(t *testing.T) {
 	bodyAtts, foundBodyAtts := body1[BodyAttachments]
 	assert.False(t, foundBodyAtts, "not expecting '_attachments' in body but found them: %v", bodyAtts)
 }
+
+func TestAllowedAttachments(t *testing.T) {
+	defer base.SetUpTestLogging(base.LevelDebug, base.KeySync)()
+
+	var tests = []struct {
+		name              string
+		inputBlipProtocol string
+		inputAttVersion   int
+	}{
+		{"TestAllowedAttachmentsCBMobile2AttVer1", BlipCBMobileReplicationV2, AttVersion1},
+		{"TestAllowedAttachmentsCBMobile2AttVer2", BlipCBMobileReplicationV2, AttVersion2},
+		{"TestAllowedAttachmentsCBMobile3AttVer1", BlipCBMobileReplicationV3, AttVersion1},
+		{"TestAllowedAttachmentsCBMobile3AttVer2", BlipCBMobileReplicationV3, AttVersion2},
+	}
+
+	// Single document associated with multiple attachments of different digests.
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			blipContext, err := NewSGBlipContext(context.TODO(), "")
+			blipContext.SupportedSubProtocols = []string{tt.inputBlipProtocol}
+			require.NoError(t, err, "Error creating SGBlipContext")
+			ctx := &BlipSyncContext{
+				blipContextDb: &Database{Ctx: context.TODO()},
+				blipContext:   blipContext,
+			}
+			meta := []AttachmentStorageMeta{
+				{digest: "digest1", version: tt.inputAttVersion},
+				{digest: "digest2", version: tt.inputAttVersion},
+			}
+			ctx.addAllowedAttachments("doc1", meta)
+			require.True(t, ctx.isAttachmentAllowed("digest1"))
+			require.True(t, ctx.isAttachmentAllowed("digest2"))
```

Comment:
In the case where blipProtocol==BlipCBMobileReplicationV3, won't attachments be stored in allowedAttachments indexed by docID+digest?  These tests don't seem to be accounting for that when calling isAttachmentAllowed.

Reply:
Makes sense, when looking at this further, it seems that the protocol version is alway returned as BlipCBMobileReplicationV2. To fix that we can make use of the NewBlipTesterFromSpecWithRT, but the  TestAllowedAttachments needs to be moved to the rest package. In addition to that, a setter method on BlipSyncContext is necessary to set the blip.Context.